src/Components/Components/src/Reflection/ComponentProperties.cs

```diff
@@ -44,6 +47,24 @@ public static void SetProperties(in ParameterView parameters, object target)
                         ThrowForUnknownIncomingParameterName(targetType, parameterName);
                         throw null; // Unreachable
                     }
+                    else if (writer.Cascading && !parameter.Cascading)
+                    {
+                        // We don't allow you to set a cascading parameter with a non-cascading value. Put another way:
+                        // cascading parameters are not part of the public API of a component, so it's not reasonable
+                        // for someone to set it directly.
+                        //
+                        // If we find a strong reason for this to work in the future we can reverse our decision since
+                        // this throws today.
+                        ThrowForSettingCascadingParameterWithNonCascadingValue(targetType, parameterName);
+                        throw null; // Unreachable
+                    }
+                    else if (!writer.Cascading && parameter.Cascading)
+                    {
+                        // We're giving a more specific error here because trying to set a non-cascading parameter
+                        // with a cascading value is likely deliberate (but not supported), or is a bug in our code.
+                        ThrowForSettingParameterWithCascadingValue(targetType, parameterName);
```

Comment:
My understanding is that this can't happen by accident (e.g., there happens by coincidence to be a CascadingValue with the same name as a non-cascading parameter), because when setting up the set of subscriptions to cascading parameters, we only pick the ones that the component asks for.

As such this code path only occurs if there's some kind of bug in the renderer (i.e., it's supplying an unwanted cascading parameter).

Assuming that's all true this is good and fine, though it might be nice to clarify the comment on L63-64 to emphasise how if this happens, it's a bug in the renderer and not in user code.

Reply:
> it's a bug in the renderer and not in user code

Or highly nonstandard user code (passing a manually-constructed `ParameterView`), or more likely, user unit test code. Right?

Reply:
Right, I'm referring to the case where a user constructed their own parameter view (`is likely deliberate (but not supported` ) whether it's in tests or not. 

I plan on leaving the comment as is - I think it's already sufficiently descriptive.